collector/common_collector.go

```diff
@@ -57,8 +57,16 @@ func parseLogService(ch chan<- prometheus.Metric, metrics map[string]Metric, sub
 	}
 	wg2 := &sync.WaitGroup{}
 	wg2.Add(len(logEntries))
+	processed := make(map[string]bool)
 	for _, logEntry := range logEntries {
+		_, exists := processed[logEntry.MessageID]
+		if exists {
+			wg2.Done()
```

Comment:
Note: no logger. We'd have to pass it in as an extra argument if we think it is worth it.

Reply:
Not to worry, we can revisit